GitUI/UserControls/FileStatusList.cs

```diff
@@ -663,7 +676,11 @@ public void SetDiffs(IReadOnlyList<GitRevision> revisions)
 
                     foreach (var rev in parentRevs)
                     {
-                        tuples.Add((rev, Module.GetDiffFilesWithSubmodulesStatus(rev.Guid, Revision.Guid, Revision.ParentIds?.FirstOrDefault()?.ToString())));
+                        var desc = (rev.ObjectId != baseRevGuid ? _diffWithParent.Text : _diffWithBase.Text)
+                                   + GetDescriptionForRevision(rev.ObjectId);
+                        tuples.Add((rev, desc, Module.GetDiffFilesWithSubmodulesStatus(rev.Guid,
+                            Revision.Guid,
+                            Revision.ParentIds?.FirstOrDefault()?.ToString())));
```

Comment:
I'm not finding this readable, there are multiple context nested within one another. 
There are still implicit allocations, so let's break it up to make it slightly more readable:
```cs
var whatever = Module.GetDiffFilesWithSubmodulesStatus(rev.Guid, Revision.Guid, Revision.ParentIds?.FirstOrDefault()?.ToString());
tuples.Add((rev, desc, whatever));
```

Reply:
ok